### PR TITLE
Add error message if old alert rule tokens are read

### PR DESCRIPTION
**Pull request by fabxc**
@grobie

### CODE REVIEW

File: promql/lex.go [hunk 1]
```diff
@@ -395,6 +405,12 @@ func (l *lexer) errorf(format string, args ...interface{}) stateFn {
 func (l *lexer) nextItem() item {
 	item := <-l.items
 	l.lastPos = item.pos
+
+	// TODO(fabxc): remove for version 1.0.
```
Comment (grobie) [inline promql/lex.go]: Could even be 0.19.0, or will that be the last 0. release \o/?

**Comment by grobie** [comment]
Awesome! :+1:

File: promql/lex.go (hunk 1 above)
Comment (fabxc) [inline promql/lex.go]: I don't see us doing any large breaking changes that justify another 0.x and would like to go straight for 1.0 afterwards.

Comment (brian-brazil) [inline promql/lex.go]: #1503 will be a large breaking change that we should complete with 1.0.

Comment (fabxc) [inline promql/lex.go]: Yes, but that's essentially the last thing and doable within the scope of one minor (which will be the transition to 1.0). No reason for 0.19.

**Pull request merged**